test/InterchainTokenFactory.js

```diff
@@ -162,6 +163,77 @@ describe('InterchainTokenFactory', () => {
         });
     });
 
+    describe('Gateway Interchain Token Factory', async () => {
+        let token, tokenId, tokenManagerAddress, symbol;
+        const tokenCap = BigInt(1e18);
+
+        async function deployToken(salt, lockUnlock = true) {
+            let tokenAddress = AddressZero;
+
+            if (lockUnlock) {
+                token = await deployContract(wallet, 'TestInterchainTokenStandard', [
+                    name,
+                    symbol,
+                    decimals,
+                    service.address,
+                    getRandomBytes32(),
+                ]);
+                tokenAddress = token.address;
+            }
+
+            const params = defaultAbiCoder.encode(
+                ['string', 'string', 'uint8', 'uint256', 'address', 'uint256'],
+                [name, symbol, decimals, 0, tokenAddress, 0],
+            );
+
+            await (await gateway.deployToken(params, getRandomBytes32())).wait();
+
+            tokenId = await service.interchainTokenId(AddressZero, salt);
+            tokenManagerAddress = await service.tokenManagerAddress(tokenId);
+
+            if (lockUnlock) {
+                await (await token.mint(wallet.address, tokenCap)).wait();
+                await (await token.setTokenId(tokenId)).wait();
+            } else {
+                tokenAddress = await gateway.tokenAddresses(symbol);
+                token = await getContractAt('IERC20', tokenAddress);
+                await await gateway.mintToken(
+                    defaultAbiCoder.encode(['string', 'address', 'uint256'], [symbol, wallet.address, tokenCap]),
+                    getRandomBytes32(),
+                );
+            }
+        }
+
+        it('Should register a token lock/unlock gateway token', async () => {
+            const salt = getRandomBytes32();
+            symbol = 'TT0';
+            await deployToken(salt, true);
+            const params = defaultAbiCoder.encode(['bytes', 'address'], ['0x', token.address]);
+
+            await expect(tokenFactory.registerGatewayToken(salt, symbol))
+                .to.emit(service, 'TokenManagerDeployed')
+                .withArgs(tokenId, tokenManagerAddress, GATEWAY, params);
+        });
+
+        it('Should register a token mint/burn gateway token', async () => {
```

Comment:
can you check unit test coverage?

Reply:
nothing changed seems affected (some deployer are not thouroughly covered it seems)